### PR TITLE
Add diligent tools 252005 

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **diligent-tools/all**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
Does someone know, what "Value 'self.version' doesn't support membership test" means?
@prince-chrismc @SSE4

**Comment by AndreyMlashkin** [comment]
@prince-chrismc @uilianries @SSE4 need some help here

**Comment by uilianries** [comment]
@AndreyMlashkin `CONAN_PKG` is almost deprecated. Please, use CMake targets generated by `CMakeDeps` if possible, otherwise, use the legacy `cmake_find_pakcage_multi`

**Comment by AndreyMlashkin** [comment]
> `CMakeDeps`

Current implementation of diligent-core does not use CMakeDeps and cmake_find_package_multi. But there is already a PR to make it use CMakeDeps

**Comment by AndreyMlashkin** [comment]
Relates
https://github.com/conan-io/conan-center-index/pull/13197

File: recipes/diligent-tools/all/conanfile.py [hunk 1]
```diff
@@ -116,8 +116,7 @@ def _configure_cmake(self):
         self._cmake.definitions["DILIGENT_BUILD_TOOLS_TESTS"] = False
         self._cmake.definitions["DILIGENT_BUILD_TOOLS_INCLUDE_TEST"] = False
         self._cmake.definitions["DILIGENT_NO_RENDER_STATE_PACKAGER"] = not self.options.with_render_state_packager
-
-        if ('api' in self.version and Version(self.version) >= Version('api.252005')) or Version(self.version) > Version('2.5.2'):
+        if 'cci' not in self.version and (('api' in self.version and Version(self.version) >= Version('api.252005')) or Version(self.version) > Version('2.5.2')):
```
Comment (prince-chrismc) [inline recipes/diligent-tools/all/conanfile.py:119]: You need to cast to str

**Review by prince-chrismc — COMMENTED** [review]
Linter flagged an issue

**Comment by AndreyMlashkin** [comment]
> @AndreyMlashkin `CONAN_PKG` is almost deprecated. Please, use CMake targets generated by `CMakeDeps` if possible, otherwise, use the legacy `cmake_find_pakcage_multi`

why diligent-core::diligent-core target is not available?
I an very confused, as I am able to build it normally on my machine...

**Comment by uilianries** [comment]
@AndreyMlashkin did you use the docker image `conanio/gcc11-ubuntu16.04:1.52.0` ?

**Comment by AndreyMlashkin** [comment]
@uilianries, thanks for the hint. In the end, old cmake version in the build image was an issue

**Comment by prince-chrismc** [comment]
Just need to delete some pdb then you are good :)

**Comment by AndreyMlashkin** [comment]
@prince-chrismc @uilianries @SSE4 
please review

**Comment by uilianries** [comment]
@AndreyMlashkin Could you please take a look on linter errors?

**Comment by AndreyMlashkin** [comment]
> @AndreyMlashkin Could you please take a look on linter errors?

I don't understand it's output:
```
  Error: Unable to locate the previous sha: 5e29af82577a27740e716a07ccb54f00e9f974ea
  Error: You seem to be missing 'fetch-depth: 0' or 'fetch-depth: 2'. See https://github.com/tj-actions/changed-files#usage
  Error: Process completed with exit code 1.
```

**Comment by uilianries** [comment]
@AndreyMlashkin Sorry, forget about, it's an infrastructure error. It will require some maintenance. Sorry for disturbing.

**Comment by prince-chrismc** [comment]
actions were revert, should be good to trigger it

**Comment by AndreyMlashkin** [comment]
@uilianries @prince-chrismc @SSE4
please review

File: recipes/diligent-tools/all/conanfile.py [hunk 2]
```diff
@@ -48,15 +63,17 @@ def configure(self):
...
-        if self.settings.compiler.cppstd:
-            tools.check_min_cppstd(self, 11)
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, 11)
         if self.options.shared:
             raise ConanInvalidConfiguration("Can't build diligent tools as shared lib")
 
+    def build_requirements(self):
+        self.tool_requires("cmake/3.24.2")
+
```
Comment (uilianries) [inline recipes/diligent-tools/all/conanfile.py:76]:
```suggestion
```

Not needed: https://github.com/DiligentGraphics/DiligentTools/blob/API252005/CMakeLists.txt#L1

File: recipes/diligent-tools/all/conanfile.py [hunk 3]
```diff
@@ -99,6 +119,13 @@ def _configure_cmake(self):
         self._cmake.definitions["DILIGENT_BUILD_TESTS"] = False
         self._cmake.definitions["DILIGENT_BUILD_TOOLS_TESTS"] = False
         self._cmake.definitions["DILIGENT_BUILD_TOOLS_INCLUDE_TEST"] = False
+        self._cmake.definitions["DILIGENT_NO_RENDER_STATE_PACKAGER"] = not self.options.with_render_state_packager
+        if 'cci' not in str(self.version) and (('api' in str(self.version) and Version(self.version) >= Version('api.252005')) or Version(self.version) > Version('2.5.2')):
```
Comment (uilianries) [inline recipes/diligent-tools/all/conanfile.py:123]: Why `Version(self.version) > Version('2.5.2')`? Is api. branch the same as that one using semver.

File: recipes/diligent-tools/all/conanfile.py (hunk 2 above)
Comment (AndreyMlashkin) [reply recipes/diligent-tools/all/conanfile.py:76]: build 9 failed before I added this line.
Now it is working

File: recipes/diligent-tools/all/conanfile.py (hunk 3 above)
Comment (AndreyMlashkin) [reply recipes/diligent-tools/all/conanfile.py:123]: Not sure, what you mean here. 
The following codeblock should be executed for versions higher than 2.5.2
api.25200X is higher

Comment (prince-chrismc) [reply recipes/diligent-tools/all/conanfile.py:123]: the release are all over the place https://github.com/DiligentGraphics/DiligentTools/tags

it's hard to say

**Comment by prince-chrismc** [comment]
https://github.com/conan-io/conan-center-index/pull/13211/files#diff-650612c1b8257930140536b11b726c7776a1f4a659ecab9dcb6b380924168688R1
is wrong

https://stackoverflow.com/a/48390363/8480874

**Comment by AndreyMlashkin** [comment]
> https://github.com/conan-io/conan-center-index/pull/13211/files#diff-650612c1b8257930140536b11b726c7776a1f4a659ecab9dcb6b380924168688R1 is wrong
> 
> https://stackoverflow.com/a/48390363/8480874

I see no reason, why. I don't want to have these targets global after all.

**Comment by AndreyMlashkin** [comment]
> https://github.com/conan-io/conan-center-index/pull/13211/files#diff-650612c1b8257930140536b11b726c7776a1f4a659ecab9dcb6b380924168688R1 is wrong
> 
> https://stackoverflow.com/a/48390363/8480874

I know, that cleaner solution would be to have multiple sub-targets in diligent-core and use them directly without defining this fake targets in diligent-tools, but it's a working approach and I don't see any reason to change it now

**Comment by AndreyMlashkin** [comment]
@jgsogo @prince-chrismc @uilianries  why is it happening?

**Comment by uilianries** [comment]
@AndreyMlashkin Please read https://github.com/conan-io/conan-center-index/pull/13461

About that, a package is missing for that specific package ID and need to be regenerated. It will be an internal task.

**Comment by AndreyMlashkin** [comment]
> About that, a package is missing for that specific package ID and need to be regenerated. It will be an internal task.

can I do something in the meantime?

**Comment by prince-chrismc** [comment]
There's a backend job that we call tapaholes which generates pages when they are missing 

You'll need to retrigger CI, close the pr wait 10s and then re-open it 🔁

**Comment by AndreyMlashkin** [comment]
@prince-chrismc it didn't help

**Comment by prince-chrismc** [comment]
Adding it to a list... these are on the rise

**Comment by AndreyMlashkin** [comment]
@prince-chrismc @uilianries 
Any news on it? Should I maybe open a PR for diligent-core manually, so the binaries are created for sure?

**Comment by prince-chrismc** [comment]
Definitely make a pr :)

**Comment by AndreyMlashkin** [comment]
blocked by https://github.com/conan-io/conan-center-index/pull/14179

**Comment by AndreyMlashkin** [comment]
@uilianries @prince-chrismc 
Finaly there is no blocker for this PR

**Comment by AndreyMlashkin** [comment]
@danimtb @SSE4 @czoido 
please approve

**Review by SSE4 — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Pull request merged**